Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabDocument.h

```diff
@@ -58,11 +58,13 @@ namespace AzToolsFramework::Prefab::PrefabConversionUtils
 
     private:
         bool ConstructInstanceFromPrefabDom(const PrefabDom& prefab);
+        // Marked const so this function can be called from other const functions. It will only operate on mutable variables.
+        bool RefreshPrefabDom() const;
         
         mutable PrefabDom m_dom;
         AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> m_instance;
         AZStd::string m_name;
-        AZStd::vector<AZ::Data::Asset<AZ::Data::AssetData>> m_referencedAssets;
+        mutable AZStd::vector<AZ::Data::Asset<AZ::Data::AssetData>> m_referencedAssets;
```

Comment:
😮

Reply:
The PrefabDocument uses lazy evaluation to avoid needlessly roundtripping to/from JSON. This does mean that some `const` functions need to to just-in-time serialization. The options to make that work is to do a `const_cast` inside the `const` functions or to mark the variables involved with `mutable`. Imho `mutable` is the lesser of these two evils because it uses a language keyword to indicate exactly what variables are breaking constness, while the `const_cast` hides that and essentially says that all member variables break constness.